lib/kitchen/driver/openstack.rb

```diff
@@ -303,14 +302,22 @@ def get_public_private_ips(server)
       end
 
       def get_ip(server)
-        unless config[:floating_ip].nil?
+        if config[:floating_ip]
           debug "Using floating ip: #{config[:floating_ip]}"
           return config[:floating_ip]
         end
+
+        # make sure we have the latest info
+        info 'Waiting for network information to be available...'
+        w = server.wait_for { reload && !addresses.empty? }
```

Comment:
I looked into `wait_for` options.

Good news: we don't have to `reload` in our block, it [does that](https://github.com/fog/fog-core/blob/v1.20.0/lib/fog/core/model.rb#L65) for us. So we can just test `addresses` and stop double reloading.

Bad news: there isn't really a way to enforce a minimum wait time unless we shove it in the block.

I feel like an arbitrary sleep should live in `#wait_for_server`. How about we repurpose `winrm_wait` as a general wait (default 0) and sleep without a condition?